core/src/main/java/org/apache/calcite/adapter/clone/ArrayTable.java

```diff
@@ -446,25 +451,30 @@ public static List asList(final Representation representation,
       return Pair.of(codes, codeValues);
     }
 
+    private static Pair<Object, @Nullable Comparable[]> unfreeze(Object value) {
```

Comment:
Maybe it is helpful to replace Object with a generic type?

Reply:
The point of the PR is nullability annotations, not code refactoring. In my opinion, `Pair` should be replaced with properly named classes.